modules/stochastic_tools/src/distributions/BoostLognormal.C

```diff
@@ -0,0 +1,32 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BoostLognormal.h"
+#include "Lognormal.h"
+
+#include "libmesh/auto_ptr.h"
+
+registerMooseObjectReplaced("StochasticToolsApp", BoostLognormal, "07/01/2020 00:00", Lognormal);
```

Comment:
The main motivation for removing the Boost distributions is that it's giving us a lot of headaches with our testing to test things both with and without Boost. The Grizzly build was broken for the last month because we were trying to figure out how to suppress tons of debug warnings that were coming from Boost. We've wasted weeks of time debugging Boost-related issues now, for what seems to me like little benefit.

I haven't done any performance benchmarking between our native distributions and the Boost ones -- I guess I assumed that the cost of evaluating distributions would be in the noise compared to everything else we do, but maybe that's not always the case. I'm curious just how much slower they are. What's the easiest way to see?  Just generate a bunch of Monte Carlo samples using a Normal distribution?

Reply:
My guess is that the main performance bottleneck is in `Normal::quantile()`, which is probably the trickiest of all of these functions, and I'm sure is implemented numerically in a way that differs from how Boost does it. Multiple distributions call this from their `quantile()` methods. Maybe we can do a little research in how to speed this up.

Reply:
I just looked at the Boost code, and it's calling a function to compute the inverse error function. Our current implementation uses a rational approximation that avoids doing that, but maybe it's slower. We are raising the same number to the same power many times in that function, which can be improved upon. I'm going to take a crack at that in another commit on this PR.

Reply:
Ah, I see why Boost created such a headache. I will try some performance testing between the two objects and see the difference because of `quantile`.

Evaluating the distribution could become a bottleneck when we are producing a bunch of samples and evaluating a fast surrogate model. But if we see only like a 10% speedup because of boost, it still might not be worth keeping boost around and our own functions are fast enough.

Reply:
OK, I just did some performance testing.  I modified one of the tests in `stochastic_tools/test/tests/samplers/monte_carlo` to generate 1e6 MC samples using a normal distribution (which exercises its `quantile()` function. If I turn off output, so that I don't count I/O time, it runs in 0.825s. I didn't compare to the Boost version because it's such a pain to re-build with Boost, but I compared it with the Uniform distribution, the quantile for which is basically free, and I get a run time of 0.774s. I have a hard time believing that if there is any increased performance in the Boost version that it will matter.

Reply:
Okay I was able to do a test between Normal with your improvements and BoostNormal. Looks like your implementation is faster than boost!